CodeJam.Main.Tests/Threading/TimeoutHelperTests.cs

```diff
@@ -74,5 +76,22 @@ public void TestAdjustTimeoutLimitInfiniteIfDefault()
 			Assert.AreEqual(dMinus1.AdjustTimeout(d1, infiniteIfDefault), d1);
 			Assert.AreEqual(dMinus1.AdjustTimeout(dMinus1, infiniteIfDefault), TimeoutHelper.InfiniteTimeSpan);
 		}
+
+		[TestCase(0, 1)]
+		[TestCase(1, 1)]
+		[TestCase(4, 8)]
+		[TestCase(6, 32)]
+		[TestCase(8, 60)]
+		[TestCase(10, 60)]
+		[TestCase(100, 60)]
+		public void TestBackoff(int retry, int resultSeconds)
+		{
+			var delay = TimeSpan.FromSeconds(1);
+			var max = TimeSpan.FromMinutes(1);
+			var timeoutSeconds = TimeoutHelper.ExponentialBackoffTimeout(retry, delay, max).TotalSeconds;
+			var range = Range.Create(resultSeconds * 0.8, resultSeconds * 1.2);
+
+			Assert.IsTrue(range.Contains(timeoutSeconds), $"{range} does not contain {timeoutSeconds}");
```

Comment:
Better to use CollectionAssert.Contains or Assert.That(, Does.Contain(..))

Reply:
Range is a special type, CodeJam.Range and it does not implement ICollection/IEnumerable.

So there's no way to use it together with NUnit assertion methods.

Reply:
... so I 've replaced it with Assert ... Is.InRange() :)